modules/p2/ting_openlist/ting_openlist.module

```diff
@@ -305,62 +257,65 @@ function ting_openlist_get_soap($wsdl = NULL, $trace = FALSE) {
  *   Response from Openlist.
  */
 function ting_openlist_request($function, array $arguments = array(), $detailed_log = FALSE) {
-  $response = FALSE;
-  $start_time = microtime(TRUE);
-
-  if ($soap_client = ting_openlist_get_soap()) {
-    try {
-      $response = call_user_func_array(
-        array(
-          $soap_client,
-          $function,
-        ),
-        $arguments
-      );
-
-      // If logging is enabled, log the function name and execution time.
-      if (variable_get('ting_openlist_log', FALSE)) {
-        $execution_time = microtime(TRUE) - $start_time;
-
-        $details = '';
-        if ($detailed_log || variable_get('ting_openlist_log_details', FALSE)) {
-          $details = "<pre>" . var_export($arguments, TRUE) . '</pre>';
-        }
+  timer_start(__FUNCTION__);
 
-        watchdog(
-          'openlist',
-          '@function (@execution_time s)!details',
-          array(
-            '@function' => $function,
-            '@execution_time' => $execution_time,
-            '!details' => $details,
-          ),
-          WATCHDOG_INFO
-        );
-      }
-    }
-    catch (Exception $e) {
-      $execution_time = microtime(TRUE) - $start_time;
-      watchdog(
-        'openlist',
-        'Error (@execution_time s).<br /><strong>Function:</strong> @function.<br /><strong>Exception:</strong>@exception<hr /><pre>!details</pre>',
-        array(
-          '@function' => $function,
-          '@execution_time' => $execution_time,
-          '@exception' => $e->getMessage(),
-          '!details' => var_export($arguments, TRUE),
-        ),
-        WATCHDOG_ERROR
-      );
-
-      throw $e;
-    }
+  $parts = array(
+    variable_get('ting_openlist_domain', TING_OPENLIST_DEFAULT_DOMAIN),
+    'json',
+    $function,
+  );
+
+  $query = array('library_code' => variable_get('ting_library_code', ''));
+
+  if (variable_get('ting_openlist_developer', FALSE)) {
+    $query['developer'] = 'on';
   }
-  else {
-    throw new Exception('No soap client available.');
+
+  $query += $arguments;
+
+  $url = url(
+    implode('/', $parts),
+    array('query' => $query)
+  );
+
+  $response = drupal_http_request($url);
+
+  if ($response->code != '200') {
+    watchdog(
+      'ting_openlist',
+      '<p><strong>Function:</strong><pre>@function</pre></p><p><strong>Arguments:</strong><pre>!arguments</pre></p><p><strong>Response:</strong><pre>!response</pre></p><p><strong>Execution time:</strong><pre>@execution_time</pre></p>',
+      array(
+        '@function' => $function,
+        '@execution_time' => timer_read(__FUNCTION__),
+        '!arguments' => htmlentities(var_export($arguments, TRUE)),
+        '!response' => htmlentities(var_export($response, TRUE)),
+      ),
+      WATCHDOG_ERROR
+    );
+
+    throw new Exception('Ting openlist error.');
```

Comment:
I would prefer not to use a generic exception here. PHP is expanding which errors cause exceptions to be thrown. Relying on the generic exception we cannot handle problems related to the OpenList integration specifically. In the SOAP implementation we at least had the [`SoapFault`](http://php.net/manual/en/class.soapfault.php) to go on.